pontoon/base/models.py

```diff
@@ -531,6 +510,23 @@ def create_user_profile(sender, instance, created, **kwargs):
 post_save.connect(create_user_profile, sender=User)
 
 
+def get_translation_dict(entity, translations_dict, i=None):
+    translation = {}
+
+    if entity in translations_dict:
+        translations = translations_dict[entity]
+        if i:
+            translations = [x for x in translations if x['plural_form'] == i]
+        translation = sorted(translations, key=lambda k: (-k['approved'], k['date']))[0]
+
+    return {
+        'fuzzy': translation.get('fuzzy', False),
```

Comment:
This is how I implemented it initially. Then this looked like a good idea to save some lines of code and keep dict structure in one place. But it's also less clear, I agree.